fru.c

```diff
@@ -170,23 +170,27 @@ static fru_field_t *fru_encode_6bit(const unsigned char *s /**< [in] Input strin
 }
 
 /**
- * Allocate a buffer and decode a 6-bit ASCII string from it
+ * Decode a 6-bit ASCII string
+ *
+ * Return false if there were errors during decoding and true otherwise.
  */
-static unsigned char *fru_decode_6bit(const fru_field_t *field)
+static bool fru_decode_6bit(uint8_t typelen, //<[in] typelen of encoded buffer
+                            const uint8_t *data, //< [in] buffer containing encoded 6bit ascii
+                            uint8_t *out, //< [out] buffer to decode into
+                            size_t out_len) //< [in] length of output buffer
```

Comment:
What was the reason for changing in-function allocation to external allocation? If there was any, I would ask you to do it in a separate commit with an explanation of the reasoning. Or maybe even a separate PR.